src/events/_listener.js

```diff
@@ -37,36 +33,52 @@ module.exports = function eventBusListener (inventory, req, res) {
 
     // @queues
     if (url === '/queues') {
-      message.arcType = 'queue'
-      let queue = get.queues(message.name)
-      message.lambda = queue
+      message.arcType = 'queues'
+      message.lambda = get.queues(message.name)
     }
     // @events
     else if (url === '/events' || url === '/') {
-      message.arcType = 'event'
-      let event = get.events(message.name)
-      message.lambda = event
+      message.arcType = 'events'
+      message.lambda = get.events(message.name)
     }
     // Who knows
-    else end
+    else end(message.name)
 
-    if (!quiet) {
-      let cleanup = JSON.stringify(JSON.parse(body), null, 2)
-      console.log(chalk.grey.dim('@' + message.arcType), chalk.green.dim(cleanup))
-    }
+    let { arcType, name, lambda, payload } = message
+    update.status(`@${arcType} ${name} received event`)
 
     if (!message.lambda) {
-      end(message.name)
+      end(name)
     }
     else {
-      // Spawn a fork of the Node process
-      let subprocess = proc.fork(join(__dirname, '_subprocess.js'))
-      subprocess.send(message)
-      subprocess.on('message', function _message (msg) {
-        if (!quiet) {
-          console.log(chalk.grey.dim(msg.text))
+      function mock () {
+        let lol = JSON.stringify(payload)
+        switch (arcType) {
+        case 'events':
+          return { Records: [ { Sns: { Message: lol } } ] } // this is fine
+        case 'queues':
+          return { Records: [ { body: lol } ] } // also fine
+        default:
+          throw ReferenceError('Unrecognized event type ' + arcType)
+        }
+      }
+
+      let event = mock()
+      invoke({ lambda, event, inventory, update }, function snap (err) {
+        if (err && err.message === 'lambda_not_found') {
+          update.warn(
+            `@${arcType} ${name} missing Lambda handler file\n` +
+            `Please create a handler file, or run [npx] arc init, or add 'autocreate true' to your project preferences file's '@create' pragma`
+          )
+        }
+        else if (err) {
+          update.status(`@${arcType} ${name} failed with ${err.stack}`)
```

Comment:
Maybe we should use `update.err` here?

Reply:
Yeah I went back and forth on this. Was thinking that because update.err is kinda scary, we should endeavor to reserve it for Architect errors, not userland code execution errors, but perhaps that's precious. wdyt?

Reply:
IMO an error's an error, but I don't have a strong opinion either way